src/rust/ide/view/src/documentation.rs

```diff
@@ -133,22 +149,54 @@ impl ViewModel {
             let processed = string.replace("\\n", "\n").replace("\"", "");
             let output = match input_type {
-                InputFormat::AST  => parser.generate_html_docs(processed),
+                InputFormat::AST       => parser.generate_html_docs(processed),
                 InputFormat::Docstring => parser.generate_html_doc_pure(processed),
             };
             let output = output?;
-            if output.is_empty() {
-                Ok(PLACEHOLDER_STR.into())
-            } else {
-                Ok(output)
-            }
+            Ok( if output.is_empty() { PLACEHOLDER_STR.into() } else { output } )
         }
     }
 
     /// Create a container for generated content and embed it with stylesheet.
     fn push_to_dom(&self, content:String) {
         let data_str = format!(r#"<div class="docVis">{}{}</div>"#,documentation_style(),content);
-        self.dom.dom().set_inner_html(&data_str)
+        self.dom.dom().set_inner_html(&data_str);
+    }
+
+    /// Append listeners to copy buttons in doc to enable copying examples.
+    /// It is possible to do it with implemented method, because get_elements_by_class_name
+    /// returns top-to-bottom sorted list of elements, as found in:
+    /// https://stackoverflow.com/questions/35525811/order-of-elements-in-document-getelementsbyclassname-array
+    fn attach_listeners_to_copy_buttons(&self) {
+        let code_blocks  = self.dom.dom().get_elements_by_class_name(CODE_BLOCK_CLASS);
+        let copy_buttons = self.dom.dom().get_elements_by_class_name(COPY_BUTTON_CLASS);
+        let closures     = (0..copy_buttons.length()).map(|i| -> Result<CodeCopyClosure,u32> {
+            let f = || -> Option<CodeCopyClosure> {
+                let copy_button = copy_buttons.get_with_index(i)?.dyn_into::<HtmlElement>().ok()?;
+                let code_block  = code_blocks.get_with_index(i)?.dyn_into::<HtmlElement>().ok()?;
+                let closure     = move |_event: MouseEvent| {
+                    let inner_code = code_block.inner_text();
+                    clipboard::write_text(inner_code);
+                };
+                let closure  = Closure::wrap(Box::new(closure).into_fn_mut());
+                let callback = closure.as_ref().unchecked_ref();
+                match copy_button.add_event_listener_with_callback("click",callback) {
+                    Ok(_)  => Some(closure),
+                    Err(e) => {
+                        error!(&self.logger,"Unable to add event listener to copy button: {e:?}");
+                        None
+                    },
+                }
+            };
+            f().ok_or(i)
+        });
+        let (closures,errors) : (Vec<_>,Vec<_>) = closures.partition(Result::is_ok);
+        let closures = closures.into_iter().map(Result::unwrap).collect::<Vec<CodeCopyClosure>>();
+        let errors   = errors.into_iter().map(Result::unwrap_err).collect::<Vec<u32>>();
```

Comment:
using `unwrap` and `unwrap_err` is not securte, as these ay panic. Of course they will not here, because the previous lines guarantees they are OK, but this is error prone - someone will refactor the previous line accidentaly and these will break. Do not use functions that may panic, do not use `unwrap*` until you really have to